lib/matplotlib/cbook.py

```diff
@@ -2385,7 +2384,7 @@ def pts_to_midstep(x, *args):
     # do normalization
     vertices = _step_validation(x, *args)
     # create the output array
-    steps = ma.zeros((vertices.shape[0], 2 * len(x)), np.float)
+    steps = np.zeros((vertices.shape[0], 2 * len(x)), np.float)
```

Comment:
This and the similar change in pts_to_poststep is changing the behavior, and I suspect it will break something.  _step_validation accepts subclasses including masked arrays.